src/main/java/seedu/address/logic/commands/UnassignTagCommand.java

```diff
@@ -0,0 +1,129 @@
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Notes;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Unassigns a tag to a contact in the address book.
+ */
+public class UnassignTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "unassign";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Unassigns a tag with a given tag name "
+            + "(case-insensitive) to a contact identified by a given index number used in the displayed "
+            + "persons list.\n"
+            + "Parameters: INDEX (must be a positive integer), TAGNAME"
+            + "Example: " + COMMAND_WORD + " 1 friends";
+
+    public static final String MESSAGE_SUCCESS = "Tag successfully unassigned: %1$s";
+    public static final String MESSAGE_NOT_TAGGED = "This tag was not assigned to this person previously";
+    public static final String MESSAGE_UNKNOWN_TAG = "Tag '%1$s' has not been created yet.";
+
+    private final String tagName;
+    private final Index targetIndex;
+
+    /**
+     * Creates an UnassignTagCommand to un-assign a {@code Tag} to a {@code Person}.
+     * @param targetIndex the index of the contact specified.
+     * @param tagName the name of the Tag.
+     */
+    public UnassignTagCommand(Index targetIndex, String tagName) {
+        this.tagName = tagName;
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Tag newTag = new Tag(tagName);
+        boolean tagHasBeenCreated = model.hasTag(newTag);
+
+        if (!tagHasBeenCreated) {
+            throw new CommandException(String.format(MESSAGE_UNKNOWN_TAG, tagName));
+        }
+
+        List<Person> lastShownList = model.getFilteredPersonList();
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(targetIndex.getZeroBased());
+
+        if (!canRemoveTag(personToEdit, newTag)) {
+            throw new CommandException(MESSAGE_NOT_TAGGED);
+        }
+
+        Person editedPerson = addTagToNewPerson(personToEdit, newTag);
+        model.setPerson(personToEdit, editedPerson);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, editedPerson));
+    }
+
+    /**
+     * Removes a specific {@code Tag} from the {@code Set<Tag>} of a {@code Person}.
+     * @param personToEdit the person to remove the tag from.
+     * @param newTag the tag to remove.
+     * @return the person with the removed tag.
+     */
+    private static Person addTagToNewPerson(Person personToEdit, Tag newTag) {
+        requireNonNull(personToEdit);
+        requireNonNull(newTag);
+        assert personToEdit != null;
+        assert newTag != null;
```

Comment:
I think having both `requireNonNull` and `assert` is redundant! If either one is null, requireNonNull would've thrown an error.

Reply:
True, I'll fix that!